### PR TITLE
[CENPLAT-24002] Add Python bindings for GridSurface

### PR DESCRIPTION
**Pull request by jsutton24**

### CODE REVIEW

File: omf-python/tests/test_grid_surface.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+from os import path
+from unittest import TestCase
+
+
+class TestGridSurface(TestCase):
+    def setUp(self) -> None:
+        self.omf_dir = path.join(path.dirname(__file__), "data")
+        self.one_of_everything = path.join(self.omf_dir, "one_of_everything.omf")
+
+    def test_should_contain_surface_geometry(self) -> None:
```
Comment (andrew-catalyst) [inline omf-python/tests/test_grid_surface.py]: Probably should add grid to the test name here

File: omf-python/tests/test_grid_surface.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        self.assertEqual(9, height_array.item_count())
+        self.assertEqual(
+            [0, 0, 0, 0, 2, 0, 0, 0, 0], reader.array_scalars(height_array)
+        )
+
+        self.assertIsInstance(grid_surface.grid, omf_python.Grid2.Tensor)
+
+        self.assertEqual([2, 2], grid_surface.grid.count())
+
+        grid = grid_surface.grid.grid
```
Comment (andrew-catalyst) [inline omf-python/tests/test_grid_surface.py]: grid.grid is a bit odd here. Maybe we don't need `PyGrid2` and just have grid_surface.grid return either a PyGrid2Regular or PyGrid2Tensor ?

Comment (jsutton24) [reply omf-python/tests/test_grid_surface.py]: That would probably be cleaner than using an enum. pyo3 doesn’t support complex enums yet, hence the awkward `.grid.grid`.

Comment (andrew-catalyst) [reply omf-python/tests/test_grid_surface.py]: Also my guess is the enum having methods might be causing the pyo3-stub-gen error

File: omf-python/src/geometry.rs [hunk 3]
```diff
@@ -72,3 +73,30 @@ impl PySurface {
...
+
+    #[getter]
+    fn grid(&self, py: Python<'_>) -> PyObject {
+        match self.0.grid {
+            omf::Grid2::Regular { .. } => PyGrid2Regular::from(self.0.grid.clone()).into_py(py),
+            omf::Grid2::Tensor { .. } => PyGrid2Tensor::from(self.0.grid.clone()).into_py(py),
+        }
+    }
+
+    #[getter]
```
Comment (andrew-catalyst) [inline omf-python/src/geometry.rs]: Maybe since this returns a copy of an array we shouldn't make this a getter. That's the current pattern.

**Comment by andrew-catalyst** [comment]
Looks good, maybe we can add some more of the function/class documentation over from Grid2::Tensor etc so there are python doc strings generated.

**Comment by jsutton24** [comment]
> Looks good, maybe we can add some more of the function/class documentation over from Grid2::Tensor etc so there are python doc strings generated.

Okay, I’ve copied over some more of the documentation.

**Review by andrew-catalyst — APPROVED** [review]

**Pull request merged**